backends-velox/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala

```diff
@@ -77,6 +74,19 @@ class HiveFileFormat(fileSinkConf: FileSinkDesc)
     throw QueryExecutionErrors.inferSchemaUnsupportedForHiveError()
   }
 
+  /**
+   * Gluten only supports convert [[InsertIntoHiveDirCommand]], see [[GlutenColumnarRules]].
+   */
+  private def isGlutenHiveWrite(sparkSession: SparkSession): Boolean = {
+    fileSinkConf.tableInfo
+      .getOutputFileFormatClassName()
+      .equals("org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat") &&
+      sparkSession.sparkContext.conf.getOption("spark.plugins")
+        .contains("io.glutenproject.GlutenPlugin") &&
+      sparkSession.sessionState.conf.getConf(HiveUtils.CONVERT_METASTORE_PARQUET) &&
```

Comment:
Do we need to add these configurations check [here](https://github.com/oap-project/gluten/blob/116d3dcdbec819d3fb06514bdd0a3ed03986652a/gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenColumnarRules.scala#L107)?

Reply:
The relation conversion is at post resolution hoc, which is always been applied before columnar rules. So we won't hit the right pattern if the config is unexpected.